src/error.rs

```diff
@@ -41,35 +41,57 @@ pub struct MegolmDecryptionError {
     /// detailed description
     #[wasm_bindgen(readonly)]
     pub description: JsString,
-    /// Withheld code if any. Only for `UnknownMessageIndex` error code
-    #[wasm_bindgen(readonly)]
-    pub maybe_withheld: Option<JsString>,
+
+    /// The withheld code, if any.
+    withheld_code: Option<WithheldCode>,
 }
 
+#[wasm_bindgen]
 impl MegolmDecryptionError {
     /// Creates generic error with description
     pub fn unable_to_decrypt(desc: String) -> Self {
         Self {
             code: DecryptionErrorCode::UnableToDecrypt,
             description: desc.into(),
-            maybe_withheld: None,
+            withheld_code: None,
         }
     }
+
+    /// Textual description of the withheld code, if any.
+    ///
+    /// Example: "The sender has disabled encrypting to unverified devices."
+    ///
+    /// `undefined` if we have not received a withheld code for the megolm
+    /// session.
+    #[wasm_bindgen(getter)]
+    pub fn maybe_withheld(&self) -> Option<String> {
```

Comment:
I think this should have a `js_name` to make it camelCase in JavaScript.  Also, maybe call it `maybe_withheld_reason` to make it less easy to confuse with the `withheld_code` function